README.md

```diff
@@ -36,12 +36,96 @@ TODO
+Prometheus ideas:
+
+#### Streaming APIs
+* Description: Currently the HTTP API sends all the data in one go and that has the chance overwhelm prometheus. To guard against this, there are limits set in Prometheus. Make APIs streaming where data is retrieved and sent in chunks rather than at once.
+* Recommended Skills: golang
+* Mentor(s): Frederic Branczyk (@brancz), Goutham V (@gouthamve)
+* Issue: https://github.com/prometheus/prometheus/issues/3690
+
+#### Add option to log slow queries
+* Description: Having something like PostgreSQL's log_min_duration_statement would be useful to debug performance problems. It would be great to collect detailed query information, like how many chunks were necessary to compute the result and how many had to be loaded from disk. Further having detailed stats like the number of series/samples/blocks touched would be great.
 * Recommended Skills: golang
-* Mentor(s): Ben Kochie (@SuperQ)
+* Mentor(s): Ben Kochie (@SuperQ), Goutham V (@gouthamve)
 * Issue: https://github.com/prometheus/prometheus/issues/1315
 
+#### Building a full integration testing environment and benchmarks
+* Description: We're not really excellent in the correctness / bug-free-ness department yet, partially because certain key components either lack tests or you'd have to run them in a real-world scenario for a while to discover certain bugs. I'm especially looking at our under-tested service discovery mechanisms here that require e.g. a Zookeeper or Consul as a dependency to test them for real. It'd be cool to have a test environment that runs a Prometheus release end-to-end (with different SD mechanisms) for a while and checks the results (evaluated expressions, alerts, etc.) for sanity.
+* Recommended Skills: golang, infrastructure management
+* Mentor(s): Brian Brazil (@brian-brazil), Conor Broderick (@Conorbro)
+* Issue: https://github.com/prometheus/prometheus/issues/3689
+
+#### Features that aid in building and testing alerting expressions
+* Description: One of the most annoying (but important) things when using Prometheus is making sure that your alerting expressions are correct (semantically, syntactically, and referring to existing template variables). It would be great to be able to assist the user with that. This would be both frontend and backend / tooling work.
+* Recommended Skills: golang, javascript
+* Mentor(s): Julius Volz (@juliusv)
+* Issue: https://github.com/prometheus/prometheus/issues/1154 https://github.com/prometheus/prometheus/issues/1695 https://github.com/prometheus/prometheus/issues/1220 https://github.com/prometheus/prometheus/issues/1219
+
+#### Tracing in Prometheus and Alertmanager
+* Description: We can use open-tracing to trace several things in both prometheus and alertmanager, for example, prometheus query tracing.
```

Comment:
IMO this is related to "Add option to log slow queries", as one method for debugging slow queries could be to use the embedded tracer.

I would change the description to:
"We have some rudimentary OpenTracing instrumentation in Prometheus; this project would be to extend this and develop a basic built-in tracing implementation capable of debugging slow running queries.  In addition, we would like to extend OpenTracing instrumentation to Alertmanager, and correlate traces between the two systems."

@brancz @Gouthamve WDYT?

Reply:
I think there is overlap, but also think that people probably want both slow query logging and tracing.